libs/execution/include/hpx/execution/apply.hpp

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2007-2015 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_EXECUTION_APPLY_APR_16_20012_0943AM)
+#define HPX_EXECUTION_APPLY_APR_16_20012_0943AM
+
+#include <hpx/config.hpp>
+#include <hpx/async_base/apply.hpp>
+#include <hpx/execution/executors/execution.hpp>
+#include <hpx/execution/executors/parallel_executor.hpp>
+#include <hpx/execution/traits/is_executor.hpp>
+#include <hpx/functional/deferred_call.hpp>
+
+#include <type_traits>
+#include <utility>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace detail {
+    ///////////////////////////////////////////////////////////////////////////
+    // Define apply() overloads for plain local functions and function objects.
+    // dispatching trait for hpx::apply
+
+    // launch a plain function/function object
+    template <typename Func, typename Enable>
+    struct apply_dispatch
+    {
+        template <typename F, typename... Ts>
+        HPX_FORCEINLINE static typename std::enable_if<
+            traits::detail::is_deferred_invocable<F, Ts...>::value, bool>::type
+        call(F&& f, Ts&&... ts)
+        {
+            parallel::execution::parallel_executor exec;
+            parallel::execution::post(
+                exec, std::forward<F>(f), std::forward<Ts>(ts)...);
```

Comment:
I think we know that `parallel_executor` implements `post`, thus we can replace this with:
```suggestion
            exec.post(std::forward<F>(f), std::forward<Ts>(ts)...);
```
There is no reason to go through the customization point again.